euclidean_gcd/euclidean_gcd.go

```diff
@@ -0,0 +1,37 @@
+package main
+
+import "fmt"
+
+// EuclideanGCD finds GCD of given numbers by iterative method
+// first : first number
+// second : second number
+// tmp : temporary variable which hold values
```

Comment:
No need for defining `tmp` as it is not a parameter and also it's use-case is pretty simple.